d2ir/compile.go

```diff
@@ -143,14 +143,33 @@ func (c *compiler) resolveSubstitutions(varsStack []*Map, node Node) {
 					c.errorf(node.LastRef().AST(), `could not resolve variable "%s"`, strings.Join(box.Substitution.IDA(), "."))
 					return
 				}
+				if box.Substitution.Spread {
+					if resolvedField.Composite == nil {
+						c.errorf(box.Substitution, "cannot spread non-map into map")
+						continue
+					}
+					// TODO arrays
+					if resolvedField.Map() != nil {
+						OverlayMap(ParentMap(node), resolvedField.Map())
+					}
+					// Remove the placeholder field
+					f := node.(*Field)
```

Comment:
Add a DeleteField method if one doesn't already exist.

Reply:
there is one but it takes an `ida`, which is much more common and useful. this is just a one-off